nmdc_runtime/site/translation/submission_portal_translator.py

```diff
@@ -668,6 +784,106 @@ def get_database(self) -> nmdc.Database:
+                if "protocol_link" in tab:
+                    protocol_link = tab.pop("protocol_link")
+                    nucleotide_sequencing_slots["protocol_link"] = nmdc.Protocol(
+                        url=protocol_link,
+                        type="nmdc:Protocol",
+                    )
+                # If model column was filled in, expand it into an nmdc:Instrument object. This is
+                # done by first checking the provided instrument mapping to see if the model is
+                # already present. If it is not, a new instrument object is created and added to the
+                # instrument_set. Currently, we only accept sequencing data in the submission portal
+                # that was generated by Illumina instruments, so the vendor is hardcoded here.
+                if "model" in tab:
+                    model = tab.pop("model")
+                    if model not in self.illumina_instrument_mapping:
+                        # If the model is not already in the mapping, create a new record for it
+                        nmdc_instrument_id = self._id_minter("nmdc:Instrument", 1)[0]
+                        database.instrument_set.append(
+                            nmdc.Instrument(
+                                id=nmdc_instrument_id,
+                                vendor=nmdc.InstrumentVendorEnum(
+                                    nmdc.InstrumentVendorEnum.illumina
+                                ),
+                                model=nmdc.InstrumentModelEnum(model),
+                                type="nmdc:Instrument",
+                            )
+                        )
+                        self.illumina_instrument_mapping[model] = nmdc_instrument_id
+                    nucleotide_sequencing_slots["instrument_used"] = (
+                        self.illumina_instrument_mapping[model]
+                    )
+                # Process the remaining columns according to the NucleotideSequencing class
+                # definition
+                nucleotide_sequencing_slots.update(
+                    self._transform_dict_for_class(tab, "NucleotideSequencing")
+                )
+                nucleotide_sequencing = nmdc.NucleotideSequencing(
+                    **nucleotide_sequencing_slots
+                )
+                database.data_generation_set.append(nucleotide_sequencing)
+
+                # Iterate over the columns that contain URLs and MD5 checksums and translate them
+                # into DataObject instances. Each of these DataObject instances will be connected
+                # to the NucleotideSequencing instance via the has_output/was_generated_by
+                # relationships.
+                url_fields = [
+                    ("read_1_url", "read_1_md5_checksum", "read_1"),
+                    ("read_2_url", "read_2_md5_checksum", "read_2"),
+                    ("interleaved_url", "interleaved_md5_checksum", "interleaved"),
+                ]
```

Comment:
I've frequently heard that [data_object_type](https://microbiomedata.github.io/nmdc-schema/data_object_type/) is an important field to capture on an `nmdc:DataObject` instance. I wonder if there's a way we can automatically infer the value of [data_object_type](https://microbiomedata.github.io/nmdc-schema/data_object_type/) given the URLs? It needs to be one of the permissible values on [FileTypeEnum](https://microbiomedata.github.io/nmdc-schema/FileTypeEnum/), so perhaps:

* `read_1_url` –> "Metagenome Raw Read 1"
* `read_2_url` –> "Metagenome Raw Read 2"
* `interleaved_url` –> "Metagenome Raw Reads"

Reply:
That's a great point about filling in the `data_object_type` slot. What you're suggesting makes perfect sense for data coming from one of the metagenome tabs. But we also have the possibility of data coming from the metatranscriptome tabs. @aclum do we just need to add three "Metatranscriptome Raw Read" permissible values to `FileTypeEnum`?

Reply:
See also: https://github.com/microbiomedata/nmdc-schema/pull/2413